src/machine/machine_nrf52xxx.go

```diff
@@ -176,39 +218,28 @@ func (a ADC) Get() uint16 {
 // SPI on the NRF.
 type SPI struct {
 	Bus *nrf.SPIM_Type
-	buf *[1]byte // 1-byte buffer for the Transfer method
 }
 
-// There are 3 SPI interfaces on the NRF528xx.
-var (
-	SPI0 = SPI{Bus: nrf.SPIM0, buf: new([1]byte)}
-	SPI1 = SPI{Bus: nrf.SPIM1, buf: new([1]byte)}
-	SPI2 = SPI{Bus: nrf.SPIM2, buf: new([1]byte)}
-)
-
 // SPIConfig is used to store config info for SPI.
 type SPIConfig struct {
 	Frequency uint32
 	SCK       Pin
 	SDO       Pin
 	SDI       Pin
 	LSBFirst  bool
-	Mode      uint8
+	Mode      SPIMode
 }
 
-// Configure is intended to setup the SPI interface.
-func (spi SPI) Configure(config SPIConfig) error {
+// Configure is intended to set up the SPI interface.
+func (spi *SPI) Configure(config SPIConfig) error {
 	// Disable bus to configure it
 	spi.Bus.ENABLE.Set(nrf.SPIM_ENABLE_ENABLE_Disabled)
 
-	// Pick a default frequency.
-	if config.Frequency == 0 {
-		config.Frequency = 4000000 // 4MHz
-	}
-
 	// set frequency
 	var freq uint32
 	switch {
+	case config.Frequency == 0: // default MCU SPI speed
+		freq = nrf.SPIM_FREQUENCY_FREQUENCY_M4
```

Comment:
I don't see this as improvement :( Previous code looks cleaner to me: separates concerns of ensuring default value and handling frequency.

Reply:
Found the separate if-statement superfluous, since we already have a flow control on the frequency.

Reply:
I find switch statements much easier to read but I think it's just a matter of personal preference. Nevertheless, we already have a switch here and that lonely if statement looks like it has rejoined it's people :smiley:

Reply:
> I find switch statements much easier to read but I think it's just a matter of personal preference. Nevertheless, we already have a switch here and that lonely if statement looks like it has rejoined it's people 😃

Yes, felt the same to me.

Reply:
Regarding switch, I must confess, I was struggling to understand what's going on at first.
I was expecting every 'case' to produce a unique value (number of unique values = number of cases in switch).
I would propose merge two cases, but then case condition becomes long and look ugly.

So I still prefer deal with default value first, then handle the value, regardless default it is or not.
It may look nice/smart to have only one switch without an extra "if" before it, because we can, but I'd say readability suffers, at least in my case.